### PR TITLE
Prevent Jvrollout realigning controls on collapse/expand

### PR DESCRIPTION
**Pull request by MHumm**
Contents of pull request #121 redone, as the other branch was deleted too early.
Fix for Mantis issue 6332.
[http://issuetracker.delphi-jedi.org/view.php?id=6332](http://issuetracker.delphi-jedi.org/view.php?id=6332)

### CODE REVIEW

File: jvcl/run/JvRollOut.pas [hunk 1]
```diff
@@ -993,39 +993,45 @@ procedure TJvCustomRollOut.SetExpandedSize(const Value: integer);
...
+      if Parent = FTopForm then
       begin
-        FTopForm.Height := FAHeight;
-        FOldWidthHeight.Y := FAHeight;
-      end
-      else
-      begin
-        FTopForm.Width := FAWidth;
-        FOldWidthHeight.X := FAWidth;
+        FTopForm.DisableAlign;
```
Comment (baka0815) [inline jvcl/run/JvRollOut.pas]: This ``DisableAlign`` is unnecessary when you're doing it in line 996 now.

File: jvcl/run/JvRollOut.pas [hunk 2]
```diff
@@ -993,39 +993,45 @@ procedure TJvCustomRollOut.SetExpandedSize(const Value: integer);
...
+        begin
+          FTopForm.Height := FAHeight;
+          FOldWidthHeight.Y := FAHeight;
+        end
+        else
+        begin
+          FTopForm.Width := FAWidth;
+          FOldWidthHeight.X := FAWidth;
+        end;
+        FTopForm.EnableAlign;
```
Comment (baka0815) [inline jvcl/run/JvRollOut.pas]: Now you have this ``EnableAlign`` and the one some lines later (around 1034) in your ``finally`` block.

Comment (MHumm) [reply jvcl/run/JvRollOut.pas]: Ok, I fixed that now. Redoing the pr was obviously done a bit in a hurry. Sorry. I really hope it is good enough for getting merged in now!

File: jvcl/run/JvRollOut.pas [hunk 3]
```diff
@@ -1451,13 +1454,19 @@ procedure TJvCustomRollOut.CheckChildVisibility;
...
 
-    for I := 0 to ControlCount - 1 do
-      if (Controls[I] is TWinControl) and (TWinControl(Controls[I]).Visible) then
-      begin
-        FChildControlVisibility.AddObject(Controls[I].Name, Controls[I]);
-        if CollapseCtrlsOnButton or (TWinControl(Controls[I]).Top > ButtonHeight) then
-          TWinControl(Controls[I]).Visible := False;
-      end;
+    DisableAlign;
+
```
Comment (obones) [inline jvcl/run/JvRollOut.pas]: To make it clearer that the `DisableAlign` is protected by the `try..finally` block, this empty line should be removed

File: jvcl/run/JvRollOut.pas [hunk 4]
```diff
@@ -1466,10 +1475,16 @@ procedure TJvCustomRollOut.CheckChildVisibility;
   begin
     if FChildControlVisibility <> nil then
     begin
-      for I := 0 to FChildControlVisibility.Count - 1 do
-        if FindChildControl(FChildControlVisibility[I]) <> nil then
-          TWinControl(FChildControlVisibility.Objects[I]).Visible := True;
-      FreeAndNil(FChildControlVisibility);
+      DisableAlign;
+
```
Comment (obones) [inline jvcl/run/JvRollOut.pas]: Same remark here

**Comment by MHumm** [comment]
Made the requested changes and comited them.

**Pull request merged**